internal/datacoord/services.go

```diff
@@ -1267,6 +1281,14 @@ func (s *Server) GetFlushState(ctx context.Context, req *datapb.GetFlushStateReq
 			if segment == nil || isFlushState(segment.GetState()) {
 				continue
 			}
+			vch := segment.SegmentInfo.GetInsertChannel()
+			cp := s.meta.GetChannelCheckpoint(vch)
```

Comment:
group segments by vchannel to avoid get cp for each segment

Reply:
OK